code/modules/multiz/movement.dm

```diff
@@ -132,7 +132,7 @@
 		return
 
 	var/turf/below = GetBelow(src)
-	if(!below)
+	if(!below || (locate(/obj/structure/ladder) in GetBelow(src)))		//Below is a ladder (needed for ladder/mobile)
```

Comment:
The result of GetBelow is already saved as below. Just use `locate(/obj/structure/ladder) in below`.

Also, perhaps this should be moved into `/mob/living/carn_fall()`? Make it return false and voila, mobs can't fall down ladders.